wfe2/wfe.go

```diff
@@ -74,16 +75,16 @@ type WebFrontEndImpl struct {
 	clk   clock.Clock
 	stats wfe2Stats
 
-	// certificateChains maps AIA issuer URLs to a slice of []byte containing a leading
+	// certificateChains maps IssuerNameIDs to slice of []byte containing a leading
 	// newline and one or more PEM encoded certificates separated by a newline,
 	// sorted from leaf to root. The first []byte is the default certificate chain,
 	// and any subsequent []byte is an alternate certificate chain.
-	certificateChains map[string][][]byte
+	certificateChains map[issuance.IssuerNameID][][]byte
```

Comment:
I wish we didn't have to rely on the order of a map to determine something critical like the default certificate chain.

Reply:
It's not really depending on the order of the map (which you actually can't depend on -- go guarantees that it will randomize the order of the key/value pairs every time you iterate through a map), it's depending on the order of the elements in the slice which is the map's value.

But yeah, it would certainly be possible to have a separate in-memory indicator for "this is the default". There are a few ways to go about it that I've thought of so far:
a) Have separate issuer-id-to-chain maps for "default" vs "alternate" chains
b) Have a separate issuer-id-to-integer-index map which says "the chain at index X is the default"
c) Make this an issuer-id-to-slice-of-struct where the struct contains both bytes and a "I'm the default" indicator
d) Probably other approaches too

Unfortunately, I'm not a big fan of any of those. I think (a) makes the code which actually uses the maps more complex; I think (b) would devolve to every entry in the "which index" map just being a zero; I think (c) has issues with multiple elements all claiming to be the default...

Definitely happy to approach this differently, I just haven't had any good ideas for how to do so.

Reply:
Thanks for the explanation! I didn't block the review on it, just something that occurred to me that would be nice if we had an elegant way to do it.